frontend.java

```diff
@@ -0,0 +1,540 @@
+				    		loggedIn = 1;
+				    		type= 1;
+				    		takeCommand();
+				    	}else if(logInTypeCom.equals("agent")){
+				    		loggedIn = 1;
+				    		type= 2;
+				    		takeCommand();
+				    	}else{
+				    		System.out.println("Invalid type, please log in");
+					    	System.exit(1);//Shouldnt exit, simply reset the command process
+				    	}
+				    }
+				}else{
+					//Logged In
+					takeCommand();
+				}
+	}
+	public static String input(){
+		BufferedReader br = new BufferedReader(new InputStreamReader(System.in)); 
+	    String command = null;
+	 
+	    //  read the username from the command-line; need to use try/catch with the
+	    //  readLine() method
+	    try {
+	    	command = br.readLine();
+	    } catch (IOException ioe) {
+	    	//Set up valid error message
+	    	System.out.println("IO error trying to read your name!");
+	        System.exit(1);//Shouldnt exit, simply reset the command process
+	    }
+	    command = command.toLowerCase();
+		return command;
+	}
+	public static void takeCommand(){
+		System.out.print("Command>");
+		String command = input();
+		if(command.equals("login")){
+			error("You are already logged in.");
+		}else if(command.equals("logout")){
+			front_logout();
+		}else if(command.equals("create")){
+			if(type == 2){
+				front_create();
+			}else{
+				error("Not logged into the proper account.");
+			}
+		}else if(command.equals("delete")){
+			if(type == 2){
+				front_delete();
+			}else{
+				error("Not logged into the proper account.");
+			}
+		}else if(command.equals("deposit")){
+			front_deposit();
+		}else if(command.equals("withdraw")){
+			front_withdraw();
+		}else if(command.equals("transfer")){
+			front_transfer();
+		}else{
+			error("Some error statement.");
+		}
+	}
+	public static void error(String errorMessage){
+		System.out.println(errorMessage);
+		takeCommand();
+	}
+	public static void front_create() {
+		// get the account number
+		// get the account name
+		// update the transaction summary file
+		System.out.print("Account Number:");
+		String accountNumber = frontend.input();
+		if(accountNumber(accountNumber)==false){
+			frontend.error("Error: Invalid account number.");
+		}else{
+			//validate the account number 
+			System.out.print("Account Name:");
+			String accountName = frontend.input();
+			if(accountName(accountName)==false){
+				frontend.error("Error: Invalid account name.");
+			}else{
+			//validate the account name 
+
+			
+				accountNumber = addZeros(accountNumber, 6);//adds zeros in front if size is less than 6
+
+				String str = "04_" + accountNumber + "_" + accountName; 
+				frontend.transactionFile.add(str);
+				//write.writeToSummaryFile(str);
+				System.out.println("Account created.");
+			}
+		}
+	}
+	public static void front_logout() {
+		frontend.loggedIn = 0;
+		//run main function on frontend.java
+		System.out.println("Logged out.");
+		
+		for (String s : frontend.transactionFile){
```

Comment:
Just something to think about when designing the software.

If the system unexpectedly crashes, you will look all the transaction history.  You should look to recover as many transactions as possible if such a situation where to occur.  One way to achieve this is via Java Signal and Exception handling.

http://www.oracle.com/technetwork/java/javase/signals-139944.html